docs/release.md

```diff
@@ -0,0 +1,34 @@
+# Release
+This document describes how to release the ClusterAPI BYOH infrastructure provider.
+
+
+### Current BYOH release: `v0.3.0`
+
+---
+##### Follow below steps to make a new release in BYOH
+1. Update `metadata.yaml` file if a new major/minor version is being released. This is not needed in case of patch release. Map the newly added version to supported ClusterAPI contract version.
+
+2. Create new tag from main branch. For example - `git tag v0.3.0`
+
+3. Push the tag to vmware-tanzu/cluster-api-provider-byoh repository (not in custom fork)
+`git push v0.3.0`
+
+4. This will trigger a new workflow in GitHub actions. Wait for the workflow to finish. This workflow will create a draft release having the contributions, changes and assets. Verify the assets, they should contain BYOH agent, cluster templates and infrastructure components.
+
+5. Next step is to release BYOH controller image.  Create controller image in local using below command
+
+   `IMG=projects.registry.vmware.com/cluster_api_provider_bringyourownhost/cluster-api-byoh-controller:v0.3.0 make docker-build`
+
+   This will create an image having name - `projects.registry.vmware.com/cluster_api_provider_bringyourownhost/cluster-api-byoh-controller`
+
+6. Next, this image needs to be pushed to  projects.registry.vmware.com harbor registry. Mostly project maintainers have access to the harbor registry, so they can be contacted to perform this step.
```

Comment:
```suggestion
6. Next, this image needs to be pushed to the `projects.registry.vmware.com` harbor registry. Most project maintainers have access to the harbor registry, so they can be contacted to perform this step.
```

Reply:
> Most project maintainers have access to the harbor registry, so they can be contacted to perform this step.

Add eg: @anusha94 and @dharmjit?

Reply:
I'd simply say `The project maintainers have access....` instead of `Most project maintainers have access....`